pkg/authorize/kubecertmanager/authorize.go

```diff
@@ -98,42 +128,139 @@ func (c *authorizer) validateIngress(cx *api.Context, ingress *extensions.Ingres
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), value,
 			fmt.Sprintf("invalid kube-cert-manager provider type: %s, expected: %s", value, class)))
 	}
-
+	
 	// @check the nginx is external
 	label = "kubernetes.io/ingress.class"
 	class = "nginx-external"
-
+	
 	if value, found := annontations[label]; !found {
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), "",
 			"the ingress does not specify a nginx class in annotations"))
 	} else if value != class {
 		return append(errs, field.Invalid(field.NewPath("annotations").Key(label), value,
 			fmt.Sprintf("invalid kube-cert-manager provider, expected '%s' for a http challenge", class)))
 	}
+	
+	errs = append(errs, c.validateIngressPointed(cx, ingress)...)
+	
+	return errs
+}
 
-	// @step: get namespace for this object
-	namespace, err := utils.GetCachedNamespace(cx.Client, cx.Cache, ingress.Namespace)
-	if err != nil {
-		return append(errs, field.InternalError(field.NewPath("namespace"), err))
+func (c *authorizer) validateCertManagerIngress(cx *api.Context, ingress *extensions.Ingress) field.ErrorList {
+	var errs field.ErrorList
+	
+	if managers := getCertManagerReferences(ingress); len(managers) !=1 || managers[0] != "cert-manager.io" {
+		return nil
 	}
-
-	enableDNSCheck := true
-	if v, found := namespace.GetAnnotations()[cx.Annotation(EnableDNSCheck)]; found {
-		if b, err := strconv.ParseBool(v); err == nil {
-			enableDNSCheck = b
+	
+	// there is at least one annotation or label prefixed with "cert-manager.io"
+	if value, found := ingress.GetAnnotations()["cert-manager.io/enabled"]; !found || value != "true" {
+	  	return append(errs, field.Invalid(field.NewPath("metadata.annotations.cert-manager.io/enabled"), "", "cert-manager.io annotations or labels are present, but annotation cert-manager.io/enabled: \"true\" is missing"))
+    }
+	
+    ingressValueValid := false
```

Comment:
formatting / indentation fix for this and the if statement above.

Reply:
sure; will run go fmt to get rid of any other formatting issues